org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/AbstractScanTest.java

```diff
@@ -432,15 +432,17 @@ private void checkRun(IRunnableDevice<ScanModel> scanner) throws Exception {
 	private IRunnableDevice<ScanModel> createTestScanner(AbstractPointsModel pmodel,
 			final ScanBean bean,
 			final IPublisher<ScanBean> publisher,
-			IScannable<?> monitor,
+			IScannable<?> monitorsPerPoint,
+			IScannable<?> monitorsPerScan,
 			IRunnableDevice<MockDetectorModel> detector) throws Exception {
-		return createTestScanner(pmodel, bean, publisher, monitor, detector, null);
+		return createTestScanner(pmodel, bean, publisher, monitorsPerPoint, monitorsPerScan, detector, null);
 	}
 
 	private IRunnableDevice<ScanModel> createTestScanner(AbstractPointsModel pmodel,
 														final ScanBean bean,
 														final IPublisher<ScanBean> publisher,
-														IScannable<?> monitor,
+														IScannable<?> monitorsPerPoint,
+														IScannable<?> monitorsPerScan,
 														IRunnableDevice<MockDetectorModel> detector,
 														String[] axes) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)